expression/expression.go

```diff
@@ -91,6 +93,38 @@ func EvalBool(expr Expression, row []types.Datum, ctx context.Context) (bool, er
 	return i != 0, nil
 }
 
+func evalExpr(expr Expression, tp types.TypeClass, row []types.Datum, sc *variable.StatementContext) (interface{}, bool, error) {
```

Comment:
Avoid using `interface{}`.
We can just return `(types.Datum, error)`.
Caller can call `d.GetInt64()`, `d.GetString()` ...
Use name `evalExprToTypeClass` is better.